tests/linen/module_test.py

```diff
@@ -1197,7 +1197,9 @@ def __call__(self, x, **sow_args):
         self.sow('intermediates', 'h', x, **sow_args)
         self.sow('intermediates', 'h', 2 * x, **sow_args)
         return 3 * x
-
+    variables = Foo().init(random.PRNGKey(0), 1)
+    # during init we should not collect intermediates by default
+    self.assertTrue('intermediates' not in variables)
```

Comment:
probably we should also have a test that if you pass `mutable=True` to `init` then indeed you do have access to the intermediates?

Reply:
done